packages/calcite-components/src/components/block/block.tsx

```diff
@@ -334,14 +382,42 @@ export class Block
     ) : null;
   }
 
+  renderIcon(iconPosition: string): VNode {
+    const { iconFlipRtl } = this;
+
+    if (iconPosition === undefined) {
+      return null;
+    }
+
+    const flipRtl =
+      iconFlipRtl === "both" || iconPosition === "iconStart"
+        ? iconFlipRtl === "start"
+        : iconFlipRtl === "end";
+
+    const iconValue = iconPosition === "iconStart" ? this.iconStart : this.iconEnd;
```

Comment:
The position values should be simplified to `start`/`end` since we know the context from the method name. You can map to the CSS class here.